core/rawdb/taiko_l1_origin.go

```diff
@@ -29,17 +29,34 @@ func l1OriginKey(blockID *big.Int) []byte {
 
 // L1Origin represents a L1Origin of a L2 block.
 type L1Origin struct {
-	BlockID       *big.Int    `json:"blockID" gencodec:"required"`
-	L2BlockHash   common.Hash `json:"l2BlockHash"`
-	L1BlockHeight *big.Int    `json:"l1BlockHeight" gencodec:"required"`
-	L1BlockHash   common.Hash `json:"l1BlockHash" gencodec:"required"`
+	BlockID       *big.Int       `json:"blockID" gencodec:"required"`
+	L2BlockHash   common.Hash    `json:"l2BlockHash"`
+	L1BlockHeight *big.Int       `json:"l1BlockHeight"`
+	L1BlockHash   common.Hash    `json:"l1BlockHash"`
+	BatchID       *big.Int       `json:"batchID"`
+	EndOfBlock    bool           `json:"endOfBlock"`
+	EndOfPreconf  bool           `json:"endOfPreconf"`
+	Preconfer     common.Address `json:"preconfer"`
 }
 
 type l1OriginMarshaling struct {
 	BlockID       *math.HexOrDecimal256
 	L1BlockHeight *math.HexOrDecimal256
 }
 
+// IsSoftblock returns true if the L1Origin is a softblock.
+func (l *L1Origin) IsSoftblock() bool {
+	if l.BatchID == nil {
+		return true
```

Comment:
Is it should return false?

Reply:
If there is no `BatchID`, then its a soft block.